### PR TITLE
Documentation for Multiple FQDN support for AMKO

### PR DESCRIPTION
**Pull request by RiyaGaniga-avi**

### CODE REVIEW

**Comment by swathinsankaran** [comment]
build amko

File: docs/multiple_fqdn.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Multiple FQDN Support in AMKO using HostRules
+
+AKO provides a CRD called [hostrule](https://github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/blob/master/docs/crds/hostrule.md) for virtual host properties. These hostrule objects can be utilised by AMKO for mutliple FQDN support and [local to global fqdn mapping](https://github.com/vmware/global-load-balancing-services-for-kubernetes/blob/master/docs/local_and_global_fqdn.md) in AMKO. <br>
+Hostrules can be configured so that a specific application can have the ability to have multiple FQDNs configured under a specific route/ingress for the child VS, instead of having to create the route multiple times.
```
Comment (swathinsankaran) [inline docs/multiple_fqdn.md:4]: please consider adding "Hostrules that can be configured **with `Aliases`**"

Comment (swathinsankaran) [inline docs/multiple_fqdn.md:4]: please consider adding "Hostrules that can be configured **with `Aliases`**"

File: docs/multiple_fqdn.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+AMKO can reduce the number of GSLB Services by grouping related(FQDN and its aliases) FQDNs together into one GSLB Service. 
+
+The functionality of AMKO changes based on how AMKO is configured.
```
Comment (swathinsankaran) [inline docs/multiple_fqdn.md:8]: the functionality changes based on how the hostrule is configured?

File: docs/multiple_fqdn.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+If aliases are repeated across clusters, AMKO will ignore such an alias. 
+
+
+2. If AMKO is configured to use custom global fqdn i.e., `useCustomGlobalFqdn: true` in gslb config object (a GSLB Service is created for a fqdn, only if a corresponding hostrule exists). [Refer this](https://github.com/vmware/global-load-balancing-services-for-kubernetes/blob/master/docs/local_and_global_fqdn.md)
+
+In such a case there can be 2 choices to the user. <br>
+*  A user can choose to include the aliases mentioned in the hostrule as domain names of the GSLB Service. <br>
+* A user can choose to ignore the aliases - default action taken by AMKO. 
+
+A flag call `includeAliases` under `gslb` section of the hostrule lets the user decide.
```
Comment (swathinsankaran) [inline docs/multiple_fqdn.md:35]: please change call to called.

File: docs/multiple_fqdn.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+
+2. If AMKO is configured to use custom global fqdn i.e., `useCustomGlobalFqdn: true` in gslb config object (a GSLB Service is created for a fqdn, only if a corresponding hostrule exists). [Refer this](https://github.com/vmware/global-load-balancing-services-for-kubernetes/blob/master/docs/local_and_global_fqdn.md)
+
+In such a case there can be 2 choices to the user. <br>
+*  A user can choose to include the aliases mentioned in the hostrule as domain names of the GSLB Service. <br>
+* A user can choose to ignore the aliases - default action taken by AMKO. 
+
+A flag call `includeAliases` under `gslb` section of the hostrule lets the user decide. 
+A sample hostrule object would look like :
```
Comment (swathinsankaran) [inline docs/multiple_fqdn.md:36]: please consider changing it to "A sample hostrule object would look like **this**:"

**Review by swathinsankaran — CHANGES_REQUESTED** [review]
LGTM, a few minor comments need to be addressed.

File: docs/multiple_fqdn.md (hunk 1 above)
Comment (RiyaGaniga-avi) [reply docs/multiple_fqdn.md]: Done

File: docs/multiple_fqdn.md (hunk 2 above)
Comment (RiyaGaniga-avi) [reply docs/multiple_fqdn.md]: Done

File: docs/multiple_fqdn.md (hunk 3 above)
Comment (RiyaGaniga-avi) [reply docs/multiple_fqdn.md]: Done

File: docs/multiple_fqdn.md (hunk 4 above)
Comment (RiyaGaniga-avi) [reply docs/multiple_fqdn.md]: Done

**Review by swathinsankaran — DISMISSED** [review]

File: docs/multiple_fqdn.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+# Multiple FQDN Support in AMKO using HostRules
+
+AKO provides a CRD called [hostrule](https://github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/blob/master/docs/crds/hostrule.md) for virtual host properties. These hostrule objects can be utilised by AMKO for mutliple FQDN support and [local to global fqdn mapping](https://github.com/vmware/global-load-balancing-services-for-kubernetes/blob/master/docs/local_and_global_fqdn.md) in AMKO. <br>
+Hostrules can be configured with `aliases` so that a specific application can have the ability to have multiple FQDNs configured under a specific route/ingress for the child VS, instead of having to create the route multiple times.
+
+AMKO can reduce the number of GSLB Services by grouping related(FQDN and its aliases) FQDNs together into one GSLB Service. 
+
+The functionality of AMKO changes based on how the hostrule is configured.
+
+1. If AMKO is configured normally i.e., `useCustomGlobalFqdn: false` in gslb config (AMKO creates a GSLB Service for every distinct ingress/route hostname).
```
Comment (akshayhavile) [inline docs/multiple_fqdn.md:10]: @RiyaGaniga-avi : `(AMKO creates a GSLB Service for every distinct ingress/route hostname)` --> this sentence should be comma separated and `(' should be removed. As this sentence represent action done on if sentence.

File: docs/multiple_fqdn.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    fqdn: foo.region1.com    #fqdn/hostname 
+    fqdnType: Exact 
+```
+
+Aliases provided must be unique in and across clusters. 
+If aliases are repeated in a cluster, AKO will reject the hostrule.
+If aliases are repeated across clusters, AMKO will ignore such an alias. 
+
+
+2. If AMKO is configured to use custom global fqdn i.e., `useCustomGlobalFqdn: true` in gslb config object (a GSLB Service is created for a fqdn, only if a corresponding hostrule exists). [Refer this](https://github.com/vmware/global-load-balancing-services-for-kubernetes/blob/master/docs/local_and_global_fqdn.md)
```
Comment (akshayhavile) [inline docs/multiple_fqdn.md:29]: @RiyaGaniga-avi : correction should be done similar to first comment.

File: docs/multiple_fqdn.md (hunk 5 above)
Comment (RiyaGaniga-avi) [reply docs/multiple_fqdn.md]: Done

**Review by swathinsankaran — APPROVED** [review]

**Review by akshayhavile — APPROVED** [review]

**Pull request merged**